lib/rules/prefer-node-protocol.js

```diff
@@ -0,0 +1,151 @@
+/**
+ * @author Yusuke Iinuma
+ * See LICENSE file in root directory for full license.
+ */
+"use strict"
+
+const isBuiltinModule = require("is-builtin-module")
+const getConfiguredNodeVersion = require("../util/get-configured-node-version")
+const getSemverRange = require("../util/get-semver-range")
+const visitImport = require("../util/visit-import")
+const visitRequire = require("../util/visit-require")
+const mergeVisitorsInPlace = require("../util/merge-visitors-in-place")
+
+const messageId = "preferNodeProtocol"
+
+module.exports = {
+    meta: {
+        docs: {
+            description:
+                "enforce using the `node:` protocol when importing Node.js builtin modules.",
+            recommended: true,
+            url: "https://github.com/eslint-community/eslint-plugin-n/blob/HEAD/docs/rules/prefer-node-protocol.md",
+        },
+        fixable: "code",
+        messages: {
+            [messageId]: "Prefer `node:{{moduleName}}` over `{{moduleName}}`.",
+        },
+        schema: [
+            {
+                type: "object",
+                properties: {
+                    version: getConfiguredNodeVersion.schema,
+                },
+                additionalProperties: false,
+            },
+        ],
+        type: "suggestion",
+    },
+    create(context) {
+        function isCallExpression(node, { name, argumentsLength }) {
+            if (node?.type !== "CallExpression") {
+                return false
+            }
+
+            if (node.optional) {
+                return false
+            }
+
+            if (node.arguments.length !== argumentsLength) {
+                return false
+            }
+
+            if (
+                node.callee.type !== "Identifier" ||
+                node.callee.name !== name
+            ) {
+                return false
+            }
+
+            return true
+        }
+
+        function isStringLiteral(node) {
+            return node?.type === "Literal" && typeof node.type === "string"
+        }
+
+        function isStaticRequire(node) {
+            return (
+                isCallExpression(node, {
+                    name: "require",
+                    argumentsLength: 1,
+                }) && isStringLiteral(node.arguments[0])
+            )
+        }
+
+        function isEnablingThisRule(context, node) {
+            const version = getConfiguredNodeVersion(context)
+
+            const supportedVersionForEsm = "^12.20.0 || >= 14.13.1"
+            // Only check Node.js version because this rule is meaningless if configured Node.js version doesn't match semver range.
+            if (!version.intersects(getSemverRange(supportedVersionForEsm))) {
+                return false
+            }
+
+            const supportedVersionForCjs = "^14.18.0 || >= 16.0.0"
+            // Only check when using `require`
+            if (
+                isStaticRequire(node.parent) &&
```

Comment:
This is probably a duplicate now, as this is covered in the `visitRequire` hook

Reply:
`moduleStyle` is available, so I refactored it. [03024dd](https://github.com/eslint-community/eslint-plugin-n/pull/183/commits/03024dd93952054a27f9930c2d03669fc65ff601)